### PR TITLE
function of config endpoint for Node API

### PR DESCRIPTION
**Pull request by henryhuang**
For issue #44 .

Added function of set GitHub endpoint for **Node API**.

I also try to added it for CLI, but my enterprise private GitHub has other authentication if use username-password mode (o-auth mode is ok), the module **ghauth** that you choose is only for username-password, so I abandoned.

### CODE REVIEW

**Comment by ungoldman** [comment]
Thank you @henryhuang, I will try to review and test this locally soon!

File: index.js [hunk 1]
```diff
@@ -62,10 +63,15 @@ function Release (options, callback) {
...
+  // set api root if options's endpoint is not empty (http(s)://hostname/api/v3/ for github enterprise), detail see https://developer.github.com/v3/enterprise/
+  if (options.endpoint) {
+    API_ROOT = options.endpoint
+  }
+
   // check if commit exists on remote
   var getCommitOptions = extend(getAuth(options), {
     method: 'GET',
-    uri: API_ROOT + format('repos/%s/%s/git/commits/%s', options.owner, options.repo, options.target_commitish)
+    uri: API_ROOT + format('repos/%s/%s/commits/%s', options.owner, options.repo, options.target_commitish)
```
Comment (ungoldman) [inline index.js]: Hi @henryhuang, I was able to review this today and have a version I'm almost ready to release. I have one question though -- why did you remove `git/` here?

Comment (henryhuang) [inline index.js]: Because this joined string can not be accessed, I found it caused by this "git" string, so I removed it , and try it on GitHub and GitHub enterprise, all of them were worked. Are you have the same problem?

Comment (ungoldman) [inline index.js]: I did some research and found that https://developer.github.com/v3/git/commits/ is distinct from https://developer.github.com/v3/repos/commits/, but both work as a commit existence check. It is odd that this would not work on Github Enterprise but should be fine for the purposes of `gh-release`.

**Comment by ungoldman** [comment]
Your changes have been merged as part of #46. Thank you!
